solr/core/src/java/org/apache/solr/search/facet/FacetField.java

```diff
@@ -139,11 +139,11 @@ public FacetProcessor<FacetField> createFacetProcessor(FacetContext fcontext) {
     // FieldType.getDocValuesType()
 
     if (!multiToken) {
-      if (mincount > 0 && prefix == null && (ntype != null || method == FacetMethod.DVHASH)) {
+      if (mincount > 0 && prefix == null && (isNumber || method == FacetMethod.DVHASH)) {
         // TODO can we auto-pick for strings when term cardinality is much greater than DocSet
         // cardinality? or if we don't know cardinality but DocSet size is very small
         return new FacetFieldProcessorByHashDV(fcontext, this, sf);
-      } else if (ntype == null) {
+      } else if (isNumber == false) {
```

Comment:
!isNumber

Reply:
Maybe you are not familiar that Lucene & Solr has recognized this style as acceptable due to its clarity